### DISCUSSION TITLE
Tabs Local Save Location

### DISCUSSION THREAD

**Discussion by ShaSha146** [XFox111/TabsAsideExtension#92] [Q&A]
Hi, extension uninstalled when I accidentally moved my clone of this repo, so I reinstalled using the clone however all my tabs disappeared. Is there somewhere where all my old tabs would have been saved; possibly in my extensions folder which I can use to recover the tabs I previously set aside?

**Comment by XFox111** [comment]
Hi, if I understand correctly, you had extension installed from extension store which you accidentally deleted and then sideloaded one from the source code of this repository

In that case, you need to reinstall the extension from the exact extension store you previously retrieved it (Google or Microsoft) and your saved tabs should appear there

The extension has built-in cloud storage, so all your saved tabs should be there

When you install extension from different sources it has different cloud storages for each, so it's safe to install multiple versions of the extension at once

**Reply by ShaSha146** [reply]
Thanks for the quick reply, actually my original installation was sideloaded and was edited to turn off the sync feature to get rid of the tab storage limit using the method here: #85, 

When I moved the folder of my sideloaded extension to a different drive the extension disappeared from edge. So, I reinstalled using the same folder that was moved to a different drive but then all my tabs disappeared.

So I'm hoping that tabs are locally saved somewhere where they can be recovered. I don't think edge got rid of local files used for the previous version of the extension because there are extension ids/folders still present that don't match any of my currently installed extensions.

**Reply by XFox111** [reply]
> When I moved the folder of my sideloaded extension to a different drive the extension disappeared from edge. So, I reinstalled using the same folder that was moved to a different drive but then all my tabs disappeared.

Have you tried to move the extension folder back and then reinstall it? Does it bring your data back?

**Reply by ShaSha146** [reply]
The drive it was originally on had to be physically removed, I made a backup of the drive and tried reinstalling using the backup of the extension folder, the files in the backup should the same as the extension folder before moving. 

Do you think creating a new drive with same drive letter as the removed drive and moving the extension folder there could help?

**Reply by XFox111** [reply]
To be honest, I don't know. It may help

You can also try to find it in `%AppData%/Local/Microsoft/Edge/User Data/%PROFILE_NAME%/Local Extension Settings` (`Sync Extension Settings` if you have used `chrome.storage.sync`). It may still be there. Though extension folders there are named after extension ID so it may be problematic to find the right one

**Reply by ShaSha146** [reply]
I tried what I mentioned before and it worked, Reassigned another drive with the same drive letter as the original drive the extension package was installed from. Then copied the backuped extension package to the reassigned drive. After closing and opening Edge the extension was back and so were my tabs. @XFox111 I think the tabs were still saved somewhere locally and restoring the extension package to where edge was looking (the original package path) allowed it to load the extension again.

**Reply by XFox111** [reply]
Oh, that's good. Now once you've accessed your data through the extension, you can run `chrome.storage.local.get(null, values => console.log(values))` on extension's background page console (go to `chrome://extensions` and click "Background page") and you will be able to see stored data

![image](https://user-images.githubusercontent.com/28831743/188980432-7717808d-c373-4b0e-949e-f5b5df980943.png)

<img width="853" alt="image" src="https://user-images.githubusercontent.com/28831743/188980752-9e74392e-02ea-474c-9766-20023b53024f.png">
